src/compute-client/src/controller/instance.rs

```diff
@@ -1560,6 +1576,15 @@ where
             return None;
         }
 
+        // Apply a global frontier update.
+        // If this is a replica-targeted subscribe, it is important that we advance the global
+        // frontier only based on responses from the targeted replica. Otherwise, another replica
+        // could advance to the empty frontier, making us drop the subscribe on the targeted
+        // replica prematurely.
+        if PartialOrder::less_than(&subscribe.frontier, &write_frontier) {
```

Comment:
Could we move this check inside `update_global_write_frontiers`? It's now kind of duplicated, because a similar check is also performed at the other call site of `update_global_write_frontiers`.

Reply:
Done. I rename the method to `maybe_update_global_write_frontiers` to make it clear that it behaves differently than `update_replica_write_frontiers` for updates that are not frontier advancements.